polaris.shopify.com/src/components/TOC/TOC.tsx

```diff
@@ -188,7 +195,11 @@ function TOC({items, collapsibleTOC = false}: Props) {
   };
 
   return (
-    <div className={classNames(styles.TOC, isNested && styles.isNested)}>
+    <div
+      className={classNames(styles.TOC, isNested && styles.isNested)}
+      onPointerEnter={handlePointerEnter}
+      onPointerLeave={handlePointerLeave}
```

Comment:
tracking keyboard+pointer events inside the TOC to prevent the `scrollLinkIntoView` running when the TOC is being scrolled independent to the Page. 

Instead, if the TOC is scrolled without a selection being made, the TOC scroll will be restored to the currentHeading position on mouseout. 

without mouse/focus tracking:


https://github.com/user-attachments/assets/58d09a14-ce5b-469f-8026-50ae1c44d6e8



with:

https://github.com/user-attachments/assets/db2b1e88-879c-477c-af62-e45df85aefed

**note:** after leaving the TOC container without making a selection, the page scrolls to the new mouse position. This feels a little janky. However, tracking the previous scroll location could add a lot of overhead. Is this situation worth solving for, or is TOC scroll followed by selection a more common use-case?